modyn/config/schema/pipeline/trigger/performance/criterion.py

```diff
@@ -0,0 +1,99 @@
+from typing import Annotated, Literal
+
+from pydantic import Field
+
+from modyn.config.schema.base_model import ModynBaseModel
+
+# -------------------------------------------------------------------------------------------------------------------- #
+#                                                 PerformanceCriterion                                                 #
+# -------------------------------------------------------------------------------------------------------------------- #
+# uses the evaluation results directory to derive a triggering decision
+
+
+class _NamedMetricPerformanceCriterion(ModynBaseModel):
+    metric: str = Field(
+        description="The metric that should be used for the comparison. Name as defined in the evaluation config."
+    )
+
+
+class StaticPerformanceThresholdCriterion(_NamedMetricPerformanceCriterion):
+    id: Literal["StaticPerformanceThresholdCriterion"] = Field("StaticPerformanceThresholdCriterion")
+    metric_threshold: float = Field(
+        0.0,
+        description=(
+            "The minimum target metric value that the model should achieve. If the performance is NOT reached "
+            "a trigger will be forced."
+        ),
+    )
+
+
+class DynamicPerformanceThresholdCriterion(_NamedMetricPerformanceCriterion):
+    """Triggers after comparison of current performance with the a rolling
+    average of historic performances after triggers."""
+
+    id: Literal["DynamicPerformanceThresholdCriterion"] = Field("DynamicPerformanceThresholdCriterion")
+    allowed_deviation: float = Field(
+        0.05,
+        description=(
+            "The allowed deviation from the expected performance. Will only trigger if the performance is "
+            "below the expected performance minus the allowed deviation."
+        ),
+    )
+
+
+# TODO: drift: rolling average + x % (instead of percentile)
```

Comment:
don't get this todo, should either expand or remove

Reply:
already part of another open PR, I'll remove it.